### PR TITLE
[Scala3] FoldingRange - tests for new syntax

### PR DESCRIPTION
**Pull request by dos65**

### CODE REVIEW

**Review by tgodzik — CHANGES_REQUESTED** [review]
Could you take a look at `scala3-example-project` - seem to have some issues there both with folding and document outline. For example:

I seem to also get:
```
Caused by: java.lang.UnsupportedOperationException: Scala212 doesn't support and types
	at scala.meta.internal.prettyprinters.TreeSyntax$SyntaxInstances.$anonfun$syntaxTree$1(TreeSyntax.scala:692)
	at scala.meta.prettyprinters.Syntax$$anon$1.apply(Syntax.scala:9)
	at scala.meta.internal.prettyprinters.TreeSyntax$.$anonfun$apply$1(TreeSyntax.scala:1281)
	at scala.meta.prettyprinters.Syntax$$anon$1.apply(Syntax.scala:9)
	at scala.meta.prettyprinters.Api$XtensionSyntax.syntax(Api.scala:11)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.apply(DocumentSymbolProvider.scala:243)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.continue$1(DocumentSymbolProvider.scala:101)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.apply(DocumentSymbolProvider.scala:122)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.continue$1(DocumentSymbolProvider.scala:101)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.newOwner$1(DocumentSymbolProvider.scala:105)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.apply(DocumentSymbolProvider.scala:177)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.continue$1(DocumentSymbolProvider.scala:101)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.apply(DocumentSymbolProvider.scala:122)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.symbols(DocumentSymbolProvider.scala:54)
	at scala.meta.internal.parsing.DocumentSymbolProvider.$anonfun$documentSymbols$1(DocumentSymbolProvider.scala:33)
	at scala.Option.map(Option.scala:230)
	at scala.meta.internal.parsing.DocumentSymbolProvider.documentSymbols(DocumentSymbolProvider.scala:32)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$documentSymbol$1(MetalsLanguageServer.scala:1306)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$apply$2(CancelTokens.scala:26)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
```

`t.body.syntax` etc seems to be the culprint. It probably uses some default syntax there.

We can do the fix in another PR for sure, but it might be related to some issues with folding I think.

**Comment by dos65** [comment]
> Could you take a look at scala3-example-project - seem to have some issues there both with folding and document outline. For example

Yes, I've seen this issue but thought that it happens only for Scala3 dependency sources. Will fix it.

File: tests/unit/src/test/resources/foldingRange-scala3/expect/Enums.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+enum Color>>region>>:
```
[the remark this answers is not in the document]
Comment (dos65) [reply tests/unit/src/test/resources/foldingRange-scala3/expect/Enums.scala:1]: In this case with an enum, it doesn't reach the threshold value (3 lines).
If you add one more item folding range will appear.

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**